.github/workflows/docker-image.yml

```
@@ -0,0 +1,43 @@
+name: build ci docker images
+
+on:
+  push:
+    # Build on any branch when docker files or this workflow changes
+    paths:
+      - 'docker/ci/**'
+      - '.github/workflows/docker-image.yml'
+  workflow_dispatch: # manual trigger option
+
+jobs:
+  build-centos7:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: login to ghcr
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      
+      # Determine branch name
+      - name: Set branch environment variables
+        id: branch_env
+        run: |
+          BRANCH_NAME=${GITHUB_REF#refs/heads/}
```

Do we want to do this the same way as the other workflow?
```suggestion
          BRANCH_NAME=${{ github.head_ref || github.ref_name }}
```

---

Yes, consistency is probably a Good Thing(TM).